### PR TITLE
Private Routes With Roles

### PR DESCRIPTION
**Pull request by jerry-cheng5**
## Notion ticket link

[Ticket Name](https://www.notion.so/uwblueprintexecs/Task-Board-db95cd7b93f245f78ee85e3a8a6a316d)

## Implementation description
* Updated private route functionality to support roles.

## Steps to test
1. Give yourself a role in your local DB (You can check the roles table for the different roles)
2. Delete existing cached user information (You can do this in dev tools. Click the 'Application' tab and go to 'Local Storage')
3. Refresh page and log back in.
4. Try to access different pages and see if you are able to access with the allowed role(s), and if you are restricted if you don't have the right role(s). (Check App.tsx) for allowed roles per page)

## What should reviewers focus on?
* Currently the roles allowed per page are just dummy data, this will need to be updated once we solidify the role restrictions per page

## Checklist
- [ ] My PR name is descriptive and in imperative tense
- [ ] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ ] I have run the appropriate linter(s)
- [ ] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/typescript/services/implementations/__tests__/userService.test.ts [hunk 1]
```diff
@@ -12,13 +12,13 @@ const testUsers = [
     firstName: "Peter",
     lastName: "Pan",
     authId: "123",
-    roleId: RoleId.Administrator,
+    role: "Administrator",
```
Comment (aanxniee) [inline backend/typescript/services/implementations/__tests__/userService.test.ts:15]: can we use Role enum that we defined in types

File: backend/typescript/services/implementations/__tests__/userService.test.ts [hunk 2]
```diff
@@ -12,13 +12,13 @@ const testUsers = [
...
   },
   {
     firstName: "Wendy",
     lastName: "Darling",
     authId: "321",
-    roleId: RoleId.Staff,
+    role: "Staff",
```
Comment (aanxniee) [inline backend/typescript/services/implementations/__tests__/userService.test.ts:21]: same here

File: backend/typescript/services/implementations/authService.ts [hunk 3]
```diff
@@ -63,7 +63,7 @@ class AuthService implements IAuthService {
           firstName: googleUser.firstName,
           lastName: googleUser.lastName,
           email: googleUser.email,
-          roleId: RoleId.Staff,
+          role: "Staff",
```
Comment (aanxniee) [inline backend/typescript/services/implementations/authService.ts:66]: same here

Comment (aanxniee) [reply backend/typescript/services/implementations/authService.ts:66]: actually this doesn't matter we're gonna deprecate google sign in later anyway

File: backend/typescript/services/implementations/userService.ts [hunk 4]
```diff
@@ -184,18 +187,37 @@ class UserService implements IUserService {
         });
       }
 
+      let roleId;
+      if (user.role) {
+        const role = await PgRole.findOne({ where: { role_name: user.role } });
```
Comment (aanxniee) [inline backend/typescript/services/implementations/userService.ts:192]: we have a role id mapping thing in types right?

**Review by aanxniee — COMMENTED** [review]
looks good for the most part, some minor nits. also, there's unused RoleId in a few files (authservice and some test files), let's get rid of those so the linter doesn't throw a warning

File: backend/typescript/services/implementations/authService.ts (hunk 3 above)
Comment (jerry-cheng5) [reply backend/typescript/services/implementations/authService.ts]: lol already did it, good catch lmao i got lazy and just used strings

**Review by aanxniee — APPROVED** [review]

**Review by sthuray — APPROVED** [review]
Looks good! Roles 1 & 2 have access to entity/create page while roles 3 & 4 don't, as expected

**Pull request merged**